_drafts/PR-blogs/planning-docs/example-code/animated-dots/original/script.js

```diff
@@ -0,0 +1,170 @@
+window.onload = init();
+function init() {
+  canvas = document.getElementById("canvas");
+  context = canvas.getContext("2d");
+  canvas.height = window.innerHeight;
+  canvas.width = window.innerWidth;
+  mouse = { x: 0, y: 0 };
+  colors = [
+    "#af0",
+    "#7CD14E",
+    "#1CDCFF",
+    "#FFFF00",
+    "#FF0000",
+    "#aee137",
+    "#bef202",
+    "#00b2ff",
+    "#7fff24",
+    "#13cd4b",
+    "#c0fa38",
+    "#f0a",
+    "#a0f",
+    "#0af",
+    "#000000",
+  ];
+  canvas.addEventListener("mousemove", MouseMove, false);
+  canvas.addEventListener("mousedown", MouseDown, false);
+  canvas.addEventListener("mouseup", MouseUp, false);
+  window.addEventListener("resize", canvasResize, false);
+  dotsHolder = [];
+  mouseMove = false;
+  mouseDown = false;
+  for (i = 0; i < 1000; i++) {
+    dotsHolder.push(new dots());
+  }
```

Comment:
Lines 3 to 34 have a lot of unrelated concepts that are all grouped together. Contained in this section are things like: 
- global state which the app mutates as it runs (mouse position, the dots themselves, the width and height of the canvas)
- immutable constants (the array of possible colors for the dots) 
- elements queried from the DOM (the canvas)
- bootstrapping code for the `canvas` API (`canvas.getContext("2d")`)
- bootstrapping code for event listeners (`mousemove`, `mousedown`)

Grouping all of these unrelated concepts together is going to confuse a first-time reader of the code. A newcomer to the file will be able to parse it more quickly if these discrete concepts are organized into groupings within the file, e.g.: 

```js
// constants
colors = [
  "#af0",
  // etc
];

// DOM elements
const canvas = document.getElementById("canvas")

// state
let mouse = { x: 0, y: 0 } 
const dotsHolder = []
canvas.height = window.innerHeight;
canvas.width = window.innerWidth;
```